torch/nativert/graph/GraphSignature.cpp

```diff
@@ -0,0 +1,479 @@
+#include <c10/util/Exception.h>
+#include <c10/util/Logging.h>
+#include <fmt/format.h>
+#include <fmt/ranges.h>
+#include <nlohmann/json.hpp>
+#include <algorithm>
+#include <array>
+#include <iostream>
+
+#include <torch/csrc/utils/generated_serialization_types.h>
+#include <torch/nativert/graph/GraphSignature.h>
+
+namespace torch::nativert {
+
+namespace {
+
+bool isSymbolicOutput(torch::_export::Argument::Tag t) {
+  switch (t) {
+    case torch::_export::Argument::Tag::AS_TENSOR:
+    case torch::_export::Argument::Tag::AS_TENSORS:
+    case torch::_export::Argument::Tag::AS_OPTIONAL_TENSORS:
+    case torch::_export::Argument::Tag::AS_SYM_BOOL:
+    case torch::_export::Argument::Tag::AS_SYM_BOOLS:
+    case torch::_export::Argument::Tag::AS_SYM_INT:
+    case torch::_export::Argument::Tag::AS_SYM_INTS:
+    case torch::_export::Argument::Tag::AS_SYM_FLOAT:
+    case torch::_export::Argument::Tag::AS_SYM_FLOATS:
+    case torch::_export::Argument::Tag::AS_CUSTOM_OBJ:
+      return true;
+    default:
+      return false;
+  }
+}
+
+std::pair<std::string, std::string> getSpecDetails(
+    const torch::_export::InputSpec& inputSpec) {
+  // Retrieve the argument name and spec tag name
+  std::string argName;
+  std::string tagName;
```

Comment:
nit: this would probably be more readable if you just returned a std::make_pair from each switch case rather than assigning to argName and tagName and then falling down to a single point of return.